Core/DistortionPointHandler.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+
+using StarlightRiver.Core;
+using StarlightRiver.Helpers;
+
+using System;
+using System.Collections.Generic;
+
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.Graphics.Effects;
+using Terraria.Graphics.Shaders;
+
+namespace StarlightRiver.Core
+{
+    public delegate float DistortionIntensityFunction(float currentIntensity, int ticksPassed);
+
+    public delegate float DistortionProgressFunction(float currentProgress, int ticksPassed);
+
+    public delegate bool DistortionActiveFunction(float currentProgress, float currentIntensity, int ticksPassed);
+
+    public class DistortionPoint
+    {
+        public Vector2 Position;
+
+        public bool Active = true;
+
+        public float Intensity;
+        public float Progress;
+
+        public int TicksPassed = 0;
+
+        public DistortionIntensityFunction UpdateIntensity;
+        public DistortionProgressFunction UpdateProgress;
+        public DistortionActiveFunction UpdateActive;
+
+        public DistortionPoint(Vector2 position, float intensity, float progress, DistortionIntensityFunction updateIntensity, DistortionProgressFunction updateProgress, DistortionActiveFunction updateActive)
+        {
+            Position = position;
+            Intensity = intensity;
+            Progress = progress;
+            UpdateIntensity = updateIntensity;
+            UpdateProgress = updateProgress;
+            UpdateActive = updateActive;
+        }
+    }
+    public class DistortionPointHandler : ModSystem
+    {
+        public static List<DistortionPoint> DistortionPoints = new List<DistortionPoint>();
```

Comment:
Distortion points are limited to 10 as per the previous discord discussion -- no? Would it not make more sense to use a fixed size array to enforce that?

Reply:
I think a list is more flexible because it can be smaller than 10.